### DISCUSSION TITLE
Solectrus unter Ubuntu mit Docker

### DISCUSSION THREAD

**Discussion by reimuell** [solectrus/solectrus#3516] [Q&A]
Hello,
I'm tiered about my Raspberry and  typing long statements in terminal window.  I have installed Ubuntu (latest plus Docker) on a Mini pc . I would like to install Solectrus / Influxdb on this machine. My question now can I install Influxdb plus INFLUXDB CLI (I really would like to have this interface to manage the DB) prior to the Solectrus Docker installation. Or is it possible to install Solectrus/Docker and after this is done INFLUXDB CLI. Thx.
---   in deutsch    -
Ich habe den Kommondazeilenmodus meines Raspberry satt. Auf eine NipoGi läuft Ubuntu neueste Version (plus Docker) , darauf will ich jetzt Solectrus mit der InfluxDB installieren. Mir ist das Interface der Datenbank wichtig da ich mir Verbesserung vor allem bei Save&Restore erhoffe. Alles was es da auf dem Raspberry gibt ist mir zu kryptisch. Kann oder muss ich die InfluxDb vor der Docker Installation machen, kommt das automatisch mit Docker? Kann man das INLUXDB auch noch nach der Docker Installation machen.
Danke Reiner Müller

**Comment by ledermann** [comment]
Aufgrund des Namens antworte ich mal auf Deutsch ;-)

Wenn du die Installation von SOLECTRUS mit dem [Konfigurator](https://configurator.solectrus.de/) erstellt hast, dann ist InfluxDB automatisch mit dabei. Ohne InfluxDB geht bei SOLECTRUS gar nichts. InfluxDB läuft, wie alles bei SOLECTRUS, in einem eigenen Docker-Container. Es muss also nicht vorher oder nachher installiert werden, sonst gehört einfach bei der Installation mit dazu (ist also in der `compose.yaml` mit aufgeführt).

Wenn du mit "Interface der Datenbank" die in InfluxDB integrierte Web-Oberfläche meinst, dann kannst du diese natürlich benutzen. Erforderlich ist nur, dass dafür Port `8086` freigegeben wird (in der `compose.yaml`). Dann kannst du mit `http://ip:8086` darauf zugreifen, während SOLECTRUS selbst weiterhin unter `http://ip:3000` erreichbar ist.

So ganz habe ich aber nicht verstanden, welche Vorteile du dir durch den Wechsel von Raspberry auf NipoGi versprichst. Der Umgang mit Docker geht grundsätzlich über die Kommandozeile, aber außer bei Installation oder manuellen Updates hat man damit eher weniger zu tun.

**Comment by reimuell** [comment]
Hallo,
danke für die Antwort.
Die Vorteile sind eher meine Unfähigkeit mich umzustellen.
Ich war mal IT'ler, SAP usw. Aber ich will immer verstehen was da passiert
und Sicherheit liegt mir auch am Herzen. Ich hatte mir von der Docker
Oberfläche (Docker CGI) und der Influx Oberfläche versprochen einfacher und
transparenter auf Tools wie Logs, Save&Restore und "was läuft da
eigentlich" versprochen. Da ist mir der Kommandozeilenmodus zu kryptisch,
ich kenne das halt anders.
Außerdem wollte ich ein Tool auf Betriebssystemebene dass sichert. Alles
was ich da in den Solectrus Diskussionen gelesen habe erschien mir zu
aufwendig, oder wie Sicherung auf Amazon ein wenig übertrieben. Auf meinem
Raspberry habe ich jetzt Raspberry Backup (GitHub) erfolgreich am laufen.
außerdem habe ich eine Wetterstation (Weewx als Software) und Homeassistent
wollte ich auch mal einsteigen. Da erschien mir der Nipogi Mini PC doch
geeigneter. Kostet auch nicht viel mehr als ein PI4.
Ich versuche jetzt mal die Methode Kopie des Verzeichnisses Solectrus auf
den Mini PC und Installation von Solectrus, wie in den Q&A beschrieben
.Hatte ich schon mal probiert ging aber schief, Solectrus hat wieder von
Stand Null begonnen.
Alle Tools setze ich dann obendrauf wenn es läuft.
Mfg
Reiner Müller


Georg Ledermann ***@***.***> schrieb am Fr., 4. Okt. 2024,
11:14:

**Reply by ledermann** [reply]
Ok, danke für die Ergänzungen. Noch ein paar Anmerkungen von mir:

- Das Frontend von InfluxDB, das über `http://ip:8086` erreichbar ist, bietet **KEIN** Backup/Restore an. Es gibt zwar die Möglichkeit, Messwerte auszugsweise per CSV herunterzuladen und auch zu importieren, aber das ist wirklich kein Backup - die resultierenden CSV-Dateien sind riesig.
- Richtige Backups mit InfluxDB können ausschließlich mit dem Kommandozeilen-Client `influx` durchgeführt werden, eine kurze Beschreibung gibt es in #3014. Das kann Basis für eigene Scripte sein.
- Ich selbst erstelle automatisierte Backups mit https://github.com/solectrus/influxdb2-s3-backup auf Amazon S3. Ja, vermutlich bin ich der einzige, der das macht - aus meiner Sicht hat das aber erhebliche Vorteile: Backups werden automatisch erstellt und liegen dann an einem externen Ort und sind entsprechend sicher. Und Amazon ist sehr günstig, die Kosten liegen im Cent-Bereich. Klar, wenn man noch gar keinen AWS-Account hat, ist das aufwändig.
- Man kann ein Backup von InfluxDB auch durch Kopieren des Volume-Ordners erstellen. Das hat aber einen großen Nachteil: InfluxDB muss **zwingend** vorher heruntergefahren werden (durch Beenden der Docker-Container), sonst entstehen inkonsistente Daten. In der Praxis ist das somit nur relevant, wenn man eine Migration durchführt, also SOLECTRUS von einem Server zu einem anderen überträgt, da werden die Container auf dem alten Server sowieso beendet.

**Comment by reimuell** [comment]
Hallo Herr Ledermann,
gerade wollte ich auf meinen neuen Mini umziehen und habe vorher Gott sei
Dank ihren Save und Restore vom Sonntag entdeckt.
Hat geklappt. Alle Daten vorhanden. Erfolgreich umgezogen.

Super.
Vielen! Dank.

Dass werde ich jetzt regelmäßig laufen lassen.

Zur Ergänzung zur Influxdb und der Oberfläche.
Ich habe jetzt mal geschaut da läuft gar kein Influx Service sonder nur
Docker.
Damit geht die UI ja nicht. Das ist wohl alles Docker gekapselt.
Die Docker UI läuft und zeigt auch den Solectrus Container.

Ich muss mich jetzt doch in Docker einlesen, hilft ja nix.
mfg
Reiner Müller




Georg Ledermann ***@***.***> schrieb am Fr., 4. Okt. 2024,
14:05:

**Reply by ledermann** [reply]
Ja, die neuen [Backup/Restore-Scripte](https://github.com/solectrus/backup-restore) sind auch für einen Umzug geeignet. Freut mich, dass es geklappt hat.

Das Frontend von InfluxDB muss man explizit freischalten durch Freigabe von Port `8086`. Also in der `compose.yaml` ergänzen und dann neu starten (mit `docker compose up -d`). Dann gelangt man über `http://ip:8086` in die Benutzeroberfläche.

Wenn der Port nicht freigegeben ist (was standardmäßig der Fall ist), geht es **nicht**. Das ist das Prinzip von Docker-Compose: Die in der `compose.yaml` definierten Services können **ausschließlich** untereinander kommunizieren, von außen ist gar nichts erreichbar - es sein denn, man richtet explizit eine Port-Freigabe ein. Auch SOLECTRUS selbst ist nur deshalb erreichbar, weil Port `3000` freigegeben ist.

Übrigens sieht man mit `docker ps`, welche Container laufen und welche Ports jeder Container freigegeben hat.

In Docker einlesen: Ja, das ist zweifelos sinnvoll :)
